File: 02/kev.rb
def d(x=@x,y=@y);@n[y] && @n[y][x];end
def run
  o=''
  File.open("kev.txt").each_line do|l|
    l.split('').each do |c|
      if c=="\n";o<<d.to_s;next;end
      (dx,dy)={U:[0,-1],D:[0,1],L:[-1,0],R:[1,0]}[c.to_sym]
      next if d(@x+dx,@y+dy).nil?;@x+=dx;@y+=dy
  end;end;o
end

@x=2;@y=2;
@n=[[nil,nil,nil],[1,2,3,nil],[4,5,6,nil],[7,8,9,nil]]
puts "Part 1: #{run}"
@n=[[nil,nil,nil],[nil,nil,1,nil],
    [nil,2,3,4,nil],[5,6,7,8,9,nil],
    [nil,'A','B','C',nil],[nil,nil,'D',nil],
    [nil,nil,nil]]
@x=2;@y=3;
puts "Part 2: #{run}"

File: 09/kev.exs
defmodule Advent9 do
  def parse([h|t]) when h == "(" do
    a = Enum.take_while(t,fn(x) -> x != ")" end)
    [{l,_},{m,_}] = Enum.join(a) |> String.split("x") |> Enum.map(fn(x) -> Integer.parse(x) end)
    the_rest = Enum.drop(t, Enum.count(a) + 1)
    (l * m) + parse(Enum.drop(the_rest, l))
  end

  def parse([]), do: 0

  def parse([_|t]), do: 1 + parse(t)
end

{:ok, file} = File.open('kev.txt')
input = String.split(IO.read(file, :line),"")
IO.puts("Part One: #{Advent9.parse(input)}")
